userspace/libsinsp/user.cpp

```diff
@@ -31,6 +31,69 @@ limitations under the License.
 #include <grp.h>
 #endif
 
+namespace {
+
+struct passwd *__getpwuid(uint32_t uid)
+{
+// See fgetpwent() feature test macros:
+// https://man7.org/linux/man-pages/man3/fgetpwent.3.html
+#if defined HAVE_PWD_H && (defined _DEFAULT_SOURCE || defined _SVID_SOURCE)
+	static std::string host_root(scap_get_host_root());
```

Comment:
I think here same comment as above applies: we could use same API to handle both cases! Note however that this function behaves in a subtle different way from the old code: we are doing a feature test even when host root is empty (ie: when we will be calling getpwuid/getgrgid), and in the else part we just return NULL. Using getpwuid api does not require any feature test macros.

Reply:
I wanted the behavior to be consistent with scap, and I explicitly wanted to return NULL when we should use fget... and we can't. This avoids reporting the container users and groups which most likely are different than the host ones.

Reply:
Mmh what i am reporting is that in this case you are returning NULL whenever 
`#if defined HAVE_PWD_H && (defined _DEFAULT_SOURCE || defined _SVID_SOURCE)` is false, even if calling just `getpwuid` or `getgrgid` does only require `HAVE_PWD_H/HAVE_GRP_H`, right?

Reply:
Make sense. I thought about it, but now I can't recall why I left it that way